### PR TITLE
PR to merge "redux-reducer-action" to the "dev"

### PR DESCRIPTION
**Pull request by AliRezaBmeDu**
In this branch, I have done the following changes:

- Create a directory that will contain all the Redux logic (/src/redux)
- Configure a Redux store (/src/redux/store.js)
- Define a slice of state for books that:
            1.  Includes an array of books (initial state: empty array)
            2.  Includes a reducer that adds a book
            3.  Includes a reducer that removes a book
- Define a slice of state for categories that:
            1. Includes an array of categories (initial state: empty array)
            2. Includes a reducer that checks the status and always returns "Under construction"

note: There was no change of the react components in this branch.

### CODE REVIEW

File: src/redux/books/booksSlice.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+      state.books = state.books.append(newBook);
+    },
+    deleteBook: (state, action) => {
+      const bookId = action.payload.id;
+      state.books = state.books.filter((item) => item.id !== bookId);
+    },
+  },
+});
+
+export default booksSlice.reducer;
```
Comment (FrederickMih) [inline src/redux/books/booksSlice.js:25]: - Nice job exporting bookSlice as reducer. I think we are equally supposed to export addBook and deleteBook as actions. **Hints** `export const { addBook, deleteBook } = bookSlice.actions` Let's kindly export them before merging our PR :smile:

File: src/redux/categories/categoriesSlice.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+const initialState = {
+  categories: [],
+  isLoading: true,
+};
+
+const categoriesSlice = createSlice({
+  name: 'books',
+  initialState,
+  reducers: {
+    checkStatus: () => 'Under construction',
```
Comment (FrederickMih) [inline src/redux/categories/categoriesSlice.js:12]: - Kindly refer to the comment above :point_up:

**Review by FrederickMih — APPROVED** [review]
## Hi :raising_hand: :raising_hand: @AliRezaBmeDu, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 🎉 

:tada: :champagne:  :tada: :champagne: 

## Highlights :green_circle: 

- Actions anf reducer for adding and removing books have been configured in Redux Store :+1: 

## Optional suggestions
**N/A**

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/redux/books/booksSlice.js (hunk 1 above)
Comment (AliRezaBmeDu) [reply src/redux/books/booksSlice.js:25]: Thank you @FrederickMih

**Pull request merged**